node/src/validator/mod.rs

```diff
@@ -72,6 +73,8 @@ pub struct Validator<N: Network, C: ConsensusStorage<N>> {
     handles: Arc<Mutex<Vec<JoinHandle<()>>>>,
     /// The shutdown signal.
     shutdown: Arc<AtomicBool>,
+    /// Keeps track of sending pings.
+    ping: Arc<Ping<N>>,
```

Comment:
do the validators use these pings? it might be handled on the BFT side already

Reply:
Yes, they ping clients using the same mechanism. `PrimaryPing` in BFT is only for talking to other validators.